test/blackbox-tests/test-cases/sandboxing.t/run.t

```diff
@@ -237,18 +236,12 @@ be copied to the build directory.
   >   (action (bash "cp %{deps} t; dune_cmd stat kind %{deps}")))
   > EOF
 
-# CR-someday amokhov: Apparently, Dune thinks that [../link] points to the build
-# directory and raises a code error. Fix this.
-
   $ cat ../link
   hi
-  $ dune build t --sandbox hardlink 2>&1 | grep 'Internal error' -A 3
-  Internal error, please report upstream including the contents of _build/log.
-  Description:
-    ("Fs_memo.Watcher.watch called on a build path",
-    { path = In_build_dir "link" })
+  $ dune build t --sandbox hardlink 2>&1
+  symbolic link
```

Comment:
I don't think this result is correct. This section is called "Sandboxing with hard links correctly resolves (follows) symbolic links", so we expect that Dune will recognise a symlink and will follow it, so the action should print "regular file" as above. 

This is not a problem of this PR, but you shouldn't remove the CR-someday: it should just change to point out that the action observed broken symlink. (It might not actually be broken but it can't possibly point to the right thing if we simply hardlinked it into the sandbox!)